wremnants/include/muon_calibration.h

```diff
@@ -355,21 +342,57 @@ class calibration_uncertainty_helper {
             std::cout << "covinvalt\n" << covinvalt << std::endl;
           }
         }
-        
+
         // protect against outliers
+        // if (weight > 0.9998 && weight < 1.0002) {cout << "smearing weight is " << weight << "covd is " << covd << std::endl;}
         res(ivar, idownup) = std::min(weight, maxWeight_);
-
       }
     }
-
+    
     return res;
   }
-
   std::shared_ptr<const HIST> hist_;
   double minGenPt_;
   double maxWeight_;
 
 };
 
+    float smearGenPt(RVec<float> cov, int charge, float pt, float theta) {
+        float sigma2_qop = cov[0];
+        double sigma2_pt = pow((pow(pt, 2) / (charge * sin(theta))), 2) * sigma2_qop;
```

Comment:
pow(x, 2) is much slower than x*x. For a calculation done per event, I think it's worth writing it out.